lib/video_info.rb

```diff
@@ -41,11 +41,30 @@ def ==(other)
     url == other.url && video_id == other.video_id
   end
 
+  @@disable_providers = []
+
+  def self.disable_providers
+    @@disable_providers
+  end
+
+  def self.disable_providers=(providers)
+    @@disable_providers = providers
+  end
+
   private
 
   def _select_provider(url, options)
     if provider_const = _providers_const.detect { |p| p.usable?(url) }
-      provider_const.new(url, options)
+      const_provider = provider_const.new(url, options)
+
+      if defined? const_provider.provider and const_provider.provider
+        if self.class.disable_providers.map(&:downcase).include? const_provider.provider.downcase
+
+          raise UrlError, "#{const_provider.provider} is disabled"
+        end
+      end
+
+      const_provider
```

Comment:
I can isolate it into a function, sure. but if I don't check that const_provider.provider is defined and not nil, a test fails:

``` ruby
....F...

Failures:

  1) VideoInfo.usable? returns true when a provider is usable
     Failure/Error: expect(VideoInfo.usable?(url)).to be_truthy
     NoMethodError:
       undefined method `provider' for true:TrueClass
     # ./lib/video_info.rb:61:in `_select_provider'
     # ./lib/video_info.rb:26:in `initialize'
     # ./lib/video_info.rb:34:in `new'
     # ./lib/video_info.rb:34:in `usable?'
     # ./spec/lib/video_info_spec.rb:61:in `block (3 levels) in <top (required)>'
```